pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go

```diff
@@ -80,6 +80,34 @@ type KeycloakAPIRealm struct {
 	// user federation at various points.
 	// +optional
 	UserFederationMappers []KeycloakAPIUserFederationMapper `json:"userFederationMappers,omitempty"`
+
+	// User registration
+	// +optional
+	RegistrationAllowed *bool `json:"registrationAllowed,omitempty"`
+	// Email as username
+	// +optional
+	RegistrationEmailAsUsername *bool `json:"registrationEmailAsUsername,omitempty"`
+	// Edit username
+	// +optional
+	EditUsernameAllowed *bool `json:"editUsernameAllowed,omitempty"`
+	// Forgot password
+	// +optional
+	ResetPasswordAllowed *bool `json:"resetPasswordAllowed,omitempty"`
+	// Remember me
+	// +optional
+	RememberMe *bool `json:"rememberMe,omitempty"`
+	// Verify email
+	// +optional
+	VerifyEmail *bool `json:"verifyEmail,omitempty"`
+	// Login with email
+	// +optional
+	LoginWithEmailAllowed *bool `json:"loginWithEmailAllowed,omitempty"`
+	// Duplicate emails
+	// +optional
+	DuplicateEmailsAllowed *bool `json:"duplicateEmailsAllowed,omitempty"`
+	// Require SSL
+	// +optional
+	SslRequired string `json:"sslRequired,omitempty"`
```

Comment:
Just a minor comment. Should this be a custom type instead of string? It's basically an enum. What's the best practise here @slaskawi?

Reply:
My reasoning was to mimic https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_realmrepresentation as closely as possible and while there are some places where enums are used this is not one of them.
If I should change it just let me know.

Reply:
I'm fine with leaving this as String. The validation will be performed on Keycloak REST level.